packages/atomic/src/components/common/tab-manager/tab-button.tsx

```diff
@@ -0,0 +1,24 @@
+import {FunctionalComponent, h} from '@stencil/core';
```

Comment:
I would not advocate to change it for this PR; However food for thoughts for everyone:

The approach of splitting functional components is mandatory when there's re-use for multiple use case (ie: we need a similar component for commerce/search/insight). Otherwise we risk introducing inconsistencies/bugs for our users.

Here however, there is currently no re-use.

There * might be * in the future, but as of today it's not the case. 
So, the code is slightly more complex (split across multiple files) for no tangible gain "today".
But, * if * we need it in the future, the work is already done, so that's nice.

So, just something to keep in mind, wonder what other think !

Reply:
I like what's done here.
Notwithstanding the reusability of the components, I think it helped a great lot to structure the code:
You have the manager that encapsulate the logic and show what it'll do in 'broad lines' without getting too much in the nitty gritty of the CSS and whatnot.
Then, we have the functional components that are dumb, and display stuff nice and proper.

I think it's nicely done!

The only point I have mixed feelings about (but nothing I'd change), is the location of the code itself. /common/ when there are no other uses than /search/ might be an overreach, especially given we can refactor them whenever we need